chord_metadata_service/chord/workflows/wdls/vcf2maf.wdl

```diff
@@ -180,40 +191,59 @@ task vcf_2_maf {
             VEP_ENSEMBL_VERSION=$(vep --help | grep "ensembl-vep" | grep -o "[0-9]*" | head -1)
 
             # Find the location of the reference assembly FASTA file for VEP
+            #  - REF_FASTA_PATH is the directory in which we search for the FASTA file.
             VEP_CACHE_PATH_SPECIES=$(echo ~{vep_species} | tr '[:upper:]' '[:lower:]')
             REF_FASTA_PATH=~{vep_cache_dir}/${VEP_CACHE_PATH_SPECIES}/${VEP_ENSEMBL_VERSION}_${assembly_id}
 
-            # The name of the FASTA file used as a reference can not be infered
-            # consistently from primitives: GRCh37 assembly has not been updated since
-            # ensembl version 75. We rely on the file actually present in the
-            # cache directory
-            # (pattern is like: "Homo_sapiens.GRCh37.75.dna.toplevel.fa.gz")
-            REF_FASTA_TOPLEVEL=$(ls ${REF_FASTA_PATH} | grep "toplevel" | head -1)
+            # The name of the FASTA file used as a reference can not be inferred consistently from primitives.
+            # We rely on the file actually present in the cache directory:
+            REF_FASTA_TOPLEVEL=$(ls ${REF_FASTA_PATH} | grep ".fa" | head -1)
 
             # Get the sample ID from the VCF file header.
-            # Here it is assumed that only one sample is present in the file.
-            # TODO: check output and syntax if multiple samples are present.
-            SAMPLE_ID=$(bcftools query -l ${g_vcf})
+            # Here it is assumed that the first sample in the file is the tumour and the second is the normal.
+            # TODO: more generalized, depending on experiment configuration. Right now, this only really works for
+            # RENATA.
 
+            n_samples=$(bcftools query -l ${g_vcf} | wc -l)
+            if [[ $n_samples != 2 ]]; then
+                echo "WARNING: Must have two samples in the VCF for vcf2maf to work. Skipping ${g_vcf}." 1>&2
+                continue
+            fi
+
+            # xargs echo -n strips the newline.
+            TUMOR_ID=$(bcftools query -l ${g_vcf} | head -1 | xargs echo -n)
+            NORMAL_ID=$(bcftools query -l ${g_vcf} | tail -1 | xargs echo -n)
+
+            # --buffer-size default is 5000. We reduce it by a lot to reduce memory usage and avoid crashes.
             perl /opt/vcf2maf.pl \
                 --input-vcf ${filtered_vcf} \
                 --output-maf ${maf} \
+                --buffer-size 500 \
+                --tmp-dir . \
+                --verbose \
                 --vep-data ~{vep_cache_dir} \
+                --species ${VEP_CACHE_PATH_SPECIES} \
                 --ref-fasta ${REF_FASTA_PATH}/${REF_FASTA_TOPLEVEL} \
+                --ncbi-build ${assembly_id} \
                 --vep-path ${VEP_PATH} \
-                --tumor-id ${SAMPLE_ID}
+                --tumor-id ${TUMOR_ID} \
+                --normal-id ${NORMAL_ID}
```

Comment:
Should these be changed to `~{}` ?

Reply:
they're references to env vars inside the wdl script, not WDL variables